src/app/header/header.component.html

```diff
@@ -6,6 +6,12 @@
 
   <span class="fill-remaining-space"></span>
 
-  <button mat-button>Sign Up</button>
-  <button mat-raised-button color="accent">Log in</button>
+
+  <button mat-button (click)="registerPage()" *ngIf="!userService.isLoggedIn()">Sign Up</button>
+  <button mat-raised-button color="accent" (click)="loginPage()" *ngIf="!userService.isLoggedIn()">Log in</button>
+
+  <p id="username" (click)="gotoProfile()" *ngIf="userService.isLoggedIn()" [(ngModel)]="usernameText" ngDefaultControl>User: {{usernameText}}</p>
```

Comment:
If you make the `user` attribute of `UserService` public, you can just make the contents of this `userService.user.username` instead of having to fetch it yourself.